src/test/ui/associated-types/associated-types-project-from-hrtb-in-fn-body.nll.stderr

```diff
@@ -19,6 +21,12 @@ LL | fn bar<'a, 'b, I : for<'x> Foo<&'x isize>>(
 ...
 LL |     let z: I::A = if cond { x } else { y };
    |                                        ^ assignment requires that `'b` must outlive `'a`
+   |
+   = help: consider adding the following bound: `'b: 'a`
+
+help: the following changes may resolve your lifetime errors
```

Comment:
This format looks a little odd.

Reply:
@varkor Can you elaborate a bit?

Reply:
It says: "the following changes" and lists nothing (or, alternatively, another nested "help"). Why not just have the `= help: `'a` and `'b` must be the same: replace one with the other` message?

Reply:
Huh? I wonder how I failed to notice that. This is a bug. It is only supposed to print that if there are multiple suggestions to make.

Also `'a: 'b` does not imply that they are the same. Rather `'a: 'b, b': 'a` implies that `'a = 'b`.

Reply:
Oh, I misread the diff and your comment. The answer is that there may be multiple changes. For example, there may be multiple bounds to add for that MIR def, which present themselves in multiple errors.

Reply:
Is there an example of that here? I didn't spot one.

Reply:
Ok, I did some refactoring, and we should now only emit the message as you suggested if there is a single suggestion. Otherwise, we have the existing scheme.

I will update the tests when they finish running.

Reply:
> Is there an example of that here? I didn't spot one.

Hmm... good point. I will add a test that exercises this. In practice, single suggestions are the most common.